### PR TITLE
ref(apple): Mention edge cases when OOM tracking is not working

### PR DESCRIPTION
**Pull request by philipphofmann**
We often get questions about OOM not working. This PR adds information on how the OOM logic is working internally.

### CODE REVIEW

File: src/platforms/apple/common/configuration/out-of-memory.mdx [hunk 1]
```diff
@@ -6,7 +6,18 @@ description: "Learn how to turn off Out Of Memory"
...
 
-When your application uses too much RAM, the operating system kills your app without a normal termination process. Your process doesn't receive a signal or any other type of information to ensure an OOM occurred. As a result, in the Apple SDK, we track out-of-memory crashes during the app start based on heuristics. If your app runs in the background, we can't reliably detect OOMs and don't report them. If you're interested in the implementation details, you can check out [the code](https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/SentryOutOfMemoryLogic.m) to find out more.
+When your application uses too much RAM, the operating system kills your app without a normal termination process. Your process doesn't receive a signal or any other type of information to ensure an OOM occurs. As a result, in the Apple SDK, we track out-of-memory crashes during the app start based on heuristics.
+The SDK checks the following statements when the user launches the app and reports an OOM if all of them are true:
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/out-of-memory.mdx:10]:
```suggestion
When a user launches the app, the SDK checks the following statements and reports an OOM if all of them are true:
```

File: src/platforms/apple/common/configuration/out-of-memory.mdx [hunk 2]
```diff
@@ -6,7 +6,18 @@ description: "Learn how to turn off Out Of Memory"
...
+
+* The app didn't crash on the previous run.
+* The app was in the foreground / active.
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/out-of-memory.mdx:13]:
```suggestion
* The app was in the foreground/active.
```

File: src/platforms/apple/common/configuration/out-of-memory.mdx [hunk 3]
```diff
@@ -6,7 +6,18 @@ description: "Learn how to turn off Out Of Memory"
...
+* The user launched the app at least once. When your app produces an OOM during the first launch, the SDK can't detect it.
+* There was no debugger attached to the process of the app. Our logic would falsely report OOMs whenever you restart the app in development.
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/out-of-memory.mdx:15]: I don't really understand how the second sentence fits. Have made a suggestion for clarity.

```suggestion
* There was no debugger attached to the process of the app. If there was, our logic would falsely report OOMs whenever you restarted the app in development.
```

File: src/platforms/apple/common/configuration/out-of-memory.mdx [hunk 4]
```diff
@@ -6,7 +6,18 @@ description: "Learn how to turn off Out Of Memory"
...
+* The OS did not update your app.
+* The user didn't update the OS of his device.
```
Comment (imatwawana) [inline src/platforms/apple/common/configuration/out-of-memory.mdx:17]: inclusive language

```suggestion
* The user didn't update the OS of their device.
```

**Review by imatwawana — APPROVED** [review]
Mostly minor smoothing and one question for clarification. Otherwise LGTM!

**Comment by philipphofmann** [comment]
Thanks @imatwawana 👏

**Pull request merged**